src/py/crankshaft/crankshaft/clustering/kmeans.py

```diff
@@ -0,0 +1,17 @@
+from sklearn.cluster import KMeans
+import plpy
+
+def kmeans(query, no_clusters, no_init=20):
+    data = plpy.execute('''select array_agg(cartodb_id order by cartodb_id) as ids,
+        array_agg(ST_X(the_geom) order by cartodb_id) xs,
```

Comment:
This will fail for null geometries as it will provide KMeans function with invalid data.